Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/appservice/file/AppServiceFileNode.java

```diff
@@ -60,11 +112,20 @@ protected void refreshItems() {
 
     @Override
     public void onNodeDblClicked(Object context) {
-        if (Objects.isNull(this.file.getContent())) {
+        final Runnable runnable = () -> {
+            this.loadFileContent(true);
+            DefaultLoader.getIdeHelper().openAppServiceFile(this.file, context);
+        };
+        final String message = String.format("fetching file %s", this.file.getName());
+        DefaultLoader.getIdeHelper().runInBackground(this.getProject(), message, false, true, null, runnable);
+    }
+
+    private synchronized Observable<byte[]> loadFileContent(boolean force) {
+        if (force || Objects.isNull(this.file.getContent())) {
             final Observable<byte[]> content = this.fileService.getFileContent(this.file.getPath());
```

Comment:
we shall not save the reference to Observable<byte[]>, it will cost a lot of memory, instead, use a temp file( delete on exit) to hold the file content

Reply:
actually, I am considering downloading content directly from server. and don't even store it temp file, since user may always want the latest content, especially for log files.